RA_Integration/RA_Integration.vcxproj

```diff
@@ -101,6 +102,7 @@
       <DebugInformationFormat>EditAndContinue</DebugInformationFormat>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <DisableSpecificWarnings>4091</DisableSpecificWarnings>
+      <LanguageStandard>stdcpp17</LanguageStandard>
```

Comment:
Are you using c++17 features? You should set this to the minimum required version.

Reply:
C++17 has a lot of stuff that makes this easier. Like instead of using const_cast on strings, you can just call data(). Plus it has more integrity checks and better exception handling. It also has parameter packs and attributes, which significantly improved logging.

The language version doesn't matter, just the SDK. The SDK is still the same using v141_xp which supports C++17.

Reply:
Another thing would be concepts (in MSVC they're just type_traits) that can be used to validate input or overloads.